src/Fhir.Anonymizer.Shared.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs

```diff
@@ -1,15 +1,32 @@
 ﻿using System;
-using System.Linq;
 using System.Security.Cryptography;
 using System.Text;
 using Hl7.FhirPath;
+using Hl7.Fhir.Model;
+using Microsoft.Extensions.Logging;
 
 namespace Fhir.Anonymizer.Core.AnonymizerConfigurations
 {
     public class AnonymizerConfigurationValidator
     {
+        private readonly ILogger _logger = AnonymizerLogging.CreateLogger<AnonymizerConfigurationValidator>();
+        
         public void Validate(AnonymizerConfiguration config)
         {
+            var currentFullVersion = ModelInfo.Version;
+            string version = currentFullVersion.Split('.')[0];
+            if (!string.IsNullOrEmpty(config.ConfigVersion))
+            {      
+                if (!string.Equals(Constants.allowedVersion[version], config.ConfigVersion))
+                {
+                    throw new AnonymizerConfigurationErrorsException($"The version of executable and configuration do not match: executable version is {Constants.allowedVersion[version]} but configuration version is {config.ConfigVersion}.");                  
```

Comment:
To be concise here: "Configuration of fhirVersion {config.FhirVersion} not supported. Expected fhirVersion: {constants.allowedVersion[version]}".